### PR TITLE
Full support for post (device) build step

### PR DESCRIPTION
**Pull request by baconpat**
* Changed `postBuild` to `postDeviceBuild` (as of now no plans to post-process agent code the same way)
* Updated test runner to look in the `.impt.project` for a `postDeviceBuild` either in the global config, or the DG specific config that corresponds to the DG specified in `.impt.test`.
* In the `postDeviceBuild` command it's possible to interpolate the path of the generated file using `${DEVICE_CODE_PATH}`

### CODE REVIEW

File: lib/Build.js [hunk 1]
```diff
@@ -44,8 +44,9 @@ const Account = require('./Account');
 const BuilderConfig = require('./util/BuilderConfig');
 const ProjectConfig = require('./util/ProjectConfig');
 const Builder = require('Builder');
+const BuilderLifecycle = require('./util/BuilderLifecycle');
```
Comment (mm-gmbd) [inline lib/Build.js:47]: I'm not really familiar with the Builder internals and why this wasn't included/used already, but I'm going to consider this okay as you've been hand testing :)

**Review by mm-gmbd — APPROVED** [review]

File: lib/util/BuilderLifecycle.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    if (!commands || commands.length == 0) {
+        return;
+    }
+
+    for (var command of commands) {
+        const finalCommand = Object.keys(commandVariables).reduce((str, variable) => {
+            return str.replace(new RegExp(`\\\${${variable}}`, RegExp.global), commandVariables[variable]);
+        }, command);
+        if (Shell.exec(finalCommand).code !== 0) {
+            UserInteractor.processError({message: "Post-build command failure."})
```
Comment (NasibMansour) [inline lib/util/BuilderLifecycle.js]: I think this is used for both "preBuild" and "postDeviceBuild", isn't this going to print "Post-build command failure." for both cases?

File: lib/Build.js (hunk 1 above)
Comment (NasibMansour) [reply lib/Build.js:47]: It looks like some of the logic was pulled into this new util file that was added, it's a lot neater!

**Review by NasibMansour — APPROVED** [review]

**Review by NasibMansour — APPROVED** [review]

File: lib/util/BuilderLifecycle.js (hunk 2 above)
Comment (baconpat) [reply lib/util/BuilderLifecycle.js]: Great catch! I pulled this common stuff out at the end and totally whiffed on the error message. I think the `executeCommands` function to take a `label` parameter or something.

File: lib/Build.js (hunk 1 above)
Comment (baconpat) [reply lib/Build.js:47]: Yes, Nasib is right. This is a new file I introduced to pull out some common code now that the post build is happening in both tests and regular deploy/run code.

**Pull request merged**